packages/xod-client/src/editor/containers/Patch/index.jsx

```diff
@@ -196,18 +197,31 @@ class Patch extends React.Component {
 
     const { currentMode } = this.state;
     const modeHandler = MODE_HANDLERS[currentMode];
+    const normalizedWheel = normalizeWheel(event);
+
+    // Most OS does not detect horizontal scroll by wheel event by default.
+    // Only MacOS works fine. So we have to do a hack for proper horizontal
+    // scrolling with pressed Shift key on all other OS: just switch X and Y.
+    const wheel =
+      detector.os.name !== 'macosx' && event.shiftKey
```

Comment:
This one feels fragile nevertheless. Can we go on without relying on OS name? For example, the following predicate looks pretty equivalent to me:

```js
event.shiftKey && normalizedWheel.pixelX === 0
```

In contrast to the current conditional, this will trigger a swap on macOS when a user hold down the Shift key and somehow scrolled strictly horizontally?! Is it ever possible? If so, what’s in the head of this xoder who holds the Shift key while scrolling with a gesture?